presets/inference/text-generation/inference_api.py

```diff
@@ -81,10 +85,47 @@ def __post_init__(self):
 model_args = asdict(args)
 model_args["local_files_only"] = not model_args.pop('allow_remote_files')
 model_pipeline = model_args.pop('pipeline')
+combination_type = model_args.pop('combination_type')
 
 app = FastAPI()
 tokenizer = AutoTokenizer.from_pretrained(**model_args)
-model = AutoModelForCausalLM.from_pretrained(**model_args)
+base_model = AutoModelForCausalLM.from_pretrained(**model_args)
+
+def list_files(directory):
+    try:
+        result = subprocess.run(['ls', directory], capture_output=True, text=True)
+        if result.returncode == 0:
+            return result.stdout.strip().split('\n')
+        else:
+            return [f"Command execution failed with return code: {result.returncode}"]
+    except Exception as e:
+        return [f"An error occurred: {str(e)}"]
+
+output = list_files(ADAPTERS_DIR)
+# [""]
+filtered_output = [s for s in output if s.strip()]
+adapters_list = [f"{ADAPTERS_DIR}/{file}" for file in filtered_output]
+filtered_adapters_list = [path for path in adapters_list if os.path.exists(os.path.join(path, "adapter_config.json"))]
+
+if len(filtered_adapters_list) == 0:
+    model = base_model
+else: 
+    adapter_names, weights= [], []
+    for adapter_path in filtered_adapters_list:
+        adapter_name = os.base_bath(adapter_path)
```

Comment:
minor bug here - I will take care of any python bugs in next PR as im changing this file currently